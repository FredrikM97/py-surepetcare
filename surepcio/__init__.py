import logging

from surepcio.security.redact import RedactSensitiveFilter
from surepcio.client import SurePetcareClient
from surepcio.household import Household

handler = logging.StreamHandler()
handler.addFilter(RedactSensitiveFilter())

logging.basicConfig(level=logging.INFO, handlers=[handler])

