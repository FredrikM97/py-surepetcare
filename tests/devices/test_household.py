import pytest
from syrupy.assertion import SnapshotAssertion

from surepcio import Household
from tests.mock_helpers import MockClient
from tests.mock_helpers import recursive_dump


@pytest.fixture
def household_file():
    return "tests/fixture/household.json"


@pytest.mark.asyncio
async def test_snapshot(snapshot: SnapshotAssertion, household_file):
    snapshot.snapshot_dir = "tests/snapshots"
    client = MockClient(fixture_file=household_file)
    household = await client.api(Household.get_households())
    assert recursive_dump(household) == snapshot
