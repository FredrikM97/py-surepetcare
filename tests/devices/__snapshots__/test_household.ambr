# serializer version: 1
# name: test_snapshot
  list([
    dict({
      'data': dict({
        'id': 1,
        'timezone': dict({
          'id': 361,
          'name': '(UTC+01:00) Central European Time (Stockholm)',
          'timezone': 'Europe/Stockholm',
          'utc_offset': 7200,
        }),
      }),
      'id': 1,
      'timezone': 'Europe/Stockholm',
    }),
    dict({
      'data': dict({
        'id': 2,
        'timezone': dict({
          'id': 361,
          'name': '(UTC+01:00) Central European Time (Stockholm)',
          'timezone': 'Europe/Stockholm',
          'utc_offset': 7200,
        }),
      }),
      'id': 2,
      'timezone': 'Europe/Stockholm',
    }),
    dict({
      'data': dict({
        'created_at': '2023-10-01T14:30:53+00:00',
        'created_user_id': 5555555,
        'id': 222527,
        'invites': list([
          dict({
            'code': 'asdf1234123123',
            'created_at': '2023-10-01T16:39:30+00:00',
            'email_address': 'user1@gmail.com',
            'id': 133372,
            'owner': True,
            'status': 1,
            'updated_at': '2023-10-01T16:41:13+00:00',
            'user': dict({
              'acceptor': dict({
                'id': 5555555,
                'name': 'User2',
              }),
              'creator': dict({
                'id': 1234566,
                'name': 'User1',
              }),
            }),
            'version': 2,
            'write': True,
          }),
        ]),
        'name': 'Hemmet',
        'share_code': 'asharecodePrivate',
        'timezone': dict({
          'id': 361,
          'name': '(UTC+01:00) Central European Time (Stockholm)',
          'timezone': 'Europe/Stockholm',
          'utc_offset': 7200,
        }),
        'timezone_id': 361,
        'updated_at': '2023-12-13T12:52:01+00:00',
        'users': list([
          dict({
            'created_at': '2023-10-01T14:30:53+00:00',
            'id': 1234566,
            'owner': True,
            'updated_at': '2023-10-01T14:30:53+00:00',
            'user': dict({
              'id': 1234566,
              'name': 'User1',
            }),
            'version': 0,
            'write': True,
          }),
          dict({
            'created_at': '2023-10-01T16:41:13+00:00',
            'id': 5555555,
            'owner': True,
            'updated_at': '2023-10-01T16:41:13+00:00',
            'user': dict({
              'id': 5555555,
              'name': 'User2',
            }),
            'version': 0,
            'write': True,
          }),
        ]),
        'version': 2,
      }),
      'id': 222527,
      'timezone': 'Europe/Stockholm',
    }),
  ])
# ---
